lib/ain-ocean/src/indexer/loan_token.rs

```
@@ -0,0 +1,171 @@
+                id: price_active_id.clone(),
+                key: oracle_price_key,
+                sort: hex::encode(ctx.block.height.to_be_bytes()),
+                active: active_price.clone(),
+                next: next_price.clone(),
+                is_live: is_live(active_price, next_price),
+                block: ctx.block.clone(),
+            };
+            services
+                .oracle_price_active
+                .by_id
+                .put(&price_active_id, &oracle_price_active)?;
+            services
+                .oracle_price_active
+                .by_key
+                .put(&oracle_price_active.key, &oracle_price_active.id)?;
+        }
+
+        Ok(())
+    }
+    fn invalidate(&self, services: &Arc<Services>, context: &Context) -> Result<()> {
+        let ticker_id = (
+            self.currency_pair.token.clone(),
+            self.currency_pair.currency.clone(),
+            context.block.height,
+        );
+        services.oracle_price_active.by_id.delete(&ticker_id)?;
+        services
+            .oracle_price_active
+            .by_key
+            .delete(&(ticker_id.0, ticker_id.1))?;
+        Ok(())
+    }
+}
+pub fn aggregated_validate(aggrigated_price: OraclePriceAggregated, context: &Context) -> bool {
+    let minimum_live_oracles: i32 = 2;
+    if (aggrigated_price.block.time - context.block.time).abs() >= 3600 {
+        return false;
+    }
+    if aggrigated_price.aggregated.oracles.active < minimum_live_oracles {
+        return false;
+    }
+
+    if aggrigated_price.aggregated.weightage <= 0 {
+        return false;
+    }
+
+    true
+}
+
+pub fn is_live(active: OraclePriceActiveActive, next: OraclePriceActiveNext) -> bool {
+    let deviation_threshold = Decimal::from_f64(0.5).unwrap_or_default();
+    let active_price = Decimal::from_str_exact(&active.amount).unwrap_or_default();
+    let next_price = Decimal::from_str_exact(&next.amount).unwrap_or_default();
```

Let's return a Result<bool> and fail instead of setting to default